docs/ProductionChecklistForScalarDLAuditor.md

```diff
@@ -0,0 +1,172 @@
+# Production checklist for ScalarDL Auditor
+
+This checklist provides recommendations when deploying ScalarDL Auditor in a production environment.
+
+## Before you begin
+
+In this checklist, we assume that you are deploying ScalarDL Auditor on a managed Kubernetes cluster, which is recommended.
+
+## Production checklist: ScalarDL Auditor
+
+The following is a checklist of recommendations when setting up ScalarDL Auditor in a production environment.
+
+### Number of pods and Kubernetes worker nodes
```

Comment:
@feeblefakie @josh-wong  
Thank you for your review!

> Pods and worker nodes (unify the current two sections, or having two sections like `Pods` and `Worker nodes`)

In this section (current `Number of pods and Kubernetes worker nodes` section), I mention both `Pods` and `Worker nodes` as follows.

```console
you should use at least three worker nodes and deploy at least three pods spread across the worker nodes
```

And, I want to tell these two things in one section to describe about availability.

Also, what I want to tell in this section is availability. So, I think we can update this section title to `Availability`.

In addition, in the current `Worker node specifications`, what I want to tell is resource estimation. So, I think we can update that section title to `Resources`.

With these in mind, I re-considered section titles as follows.

* Checklist for ScalarDL Ledger
  * Production checklist: ScalarDL Ledger
    * Availability (old section name is `Number of pods and Kubernetes worker nodes`)
    * Resources (old section name is `Worker node specifications`)
    * Network
    * Monitoring and logging
    * Backup and restore
  * Production checklist: Client applications that access ScalarDL Ledger
    * Client application deployment (old section name is `Do not deploy your application on the same Kubernetes cluster as ScalarDL Ledger`)
    * Contract and Function (old section name is `Check if your contract and function follow guidelines`)
    * Contract versioning (old section name is `Versioning contracts`)
    * Contract limitations (old section name is `Use the same binary name, package name, and class name when you register the contract`)
    * Private key and certificate (old section name is `Private key and certificate requirements`)
    * Exception handling
* Checklist for ScalarDL Auditor
  * Production checklist: ScalarDL Auditor
    * Availability (old section name is `Number of pods and Kubernetes worker nodes`)
    * Resources (old section name is `Worker node specifications`)
    * Network
    * Monitoring and logging
    * Backup and restore
    * ScalarDL Auditor deployment (old section name is `Do not deploy ScalarDL Auditor on the same Kubernetes cluster as ScalarDL Ledger`)
    * Connection between ScalarDL Ledger and ScalarDL Auditor (old section name is `Connect ScalarDL Ledger and ScalarDL Auditor`)
    * Private key and certificate (old section name is `Private key and certificate requirements`)
  * Production checklist: Client applications that access ScalarDL Auditor
    * Client application deployment (old section name is `Do not deploy your application on the same Kubernetes cluster as ScalarDL`)
    * Client application checklist (old section name is `Checklist for the client applications that access ScalarDL Ledger`)

What do you think?
(For consistency, we have to update the checklist of ScalarDB, but I will do it in another PR after this PR is merged.)

Reply:
Sorry, I didn't see this comment until @kota2and3kan's reply😔

Yes, after coming back to this, I can see how we can improve the headings. I agree with the changes @kota2and3kan suggested, with some minor changes for clarity:

**Checklist for ScalarDL Ledger**

- For `Availability`, I feel that `Pods and worker nodes` might be more accurate for this section. But maybe I'm not understanding this section clearly. 
  - If `Availability` is indeed a better phrase for what we're explaining in this part, how about `Cluster availability` to provide a little more context?
- `Contract and Function` -> `Contract and function`

**Checklist for ScalarDL Auditor**

- For `Availability`, I feel that `Pods and worker nodes` might be more accurate for this section. But maybe I'm not understanding this section clearly. 
  - If `Availability` is indeed a better phrase for what we're explaining in this part, how about `Cluster availability` to provide a little more context?

Reply:
@josh-wong  
Thank you for your comment!

Strictly, the `availability` in this section means two things:

1. Kubernetes cluster's availability.
2. Scalar products' (i.e., ScalarDL's) availability.

Regarding `1.`, `Cluster availability` makes sense for me because it can indicate `Kubernetes cluster availability`.

However, regarding `2.`, I think `Cluster availability` is not good. This is because we don't call `ScalarDL` as `ScalarDL cluster`.

So, how about `Availability of Kubernetes and ScalarDL`?

---

As another suggestion, I think the simple title `ScalarDL availability` is also good.

The `Kubernetes availability` is a base of the `ScalarDL availability`. In other words, to keep `ScalarDL availability` is our final goal. To achieve that, we have to keep `Kubernetes availability` as a platform availability.

So, I think we can focus on our final goal `ScalarDL availability` in the section title.

I think `ScalarDL availability` is better. 

What do you think?

Reply:
@kota2and3kan Ah, thank you! I understand now. I was only considering `1.`

> The `Kubernetes availability` is a base of the `ScalarDL availability`. In other words, to keep `ScalarDL availability` is our final goal. To achieve that, we have to keep `Kubernetes availability` as a platform availability.
> 
> So, I think we can focus on our final goal `ScalarDL availability` in the section title.

I agree that `ScalarDL availability` would be better now that I understand we're concerned with ensuring that ScalarDL, which in essence means Kubernetes the platform, can withstand failures. I'm OK with that wording👍

Reply:
@feeblefakie @josh-wong  
I updated the headings in 3c3b71594d5d5f8df28a738dd348d07a10a2e500 based on the discussion in this thread!
Please take a look when you have time!

Reply:
I feel `Availability` and `ScalarDL Availability` sound a bit inconsistent with the others to me.
(I feel, if we name it Availability, then the others should be like `Observability` for monitoring and logging and `Durability` for backup and restore.)

But, I'm fine with it for now, and we can rethink about it later on.

Reply:
@feeblefakie 
Thank you for your comment!

> I feel `Availability` and `ScalarDL Availability` sound a bit inconsistent with the others to me.
(I feel, if we name it Availability, then the others should be like `Observability` for monitoring and logging and `Durability` for backup and restore.)

I see. Indeed, `Observability` looks more consistent title name.

> But, I'm fine with it for now, and we can rethink about it later on.

OK. Let's rethink/discuss it later!